### PR TITLE
Add dashboard view

### PR DESCRIPTION
**Pull request by jakub-lotko**
- add application status "advanced" enum
- use dashboard query to fetch data

### CODE REVIEW

File: src/HE.InvestmentLoans.BusinessLogic/Application/Repositories/LoanApplicationRepository.cs [hunk 1]
```diff
@@ -87,6 +87,8 @@ public void Save(LoanApplicationViewModel loanApplication, UserAccount userAccou
             name = loanApplication.Account.RegisteredName,
             contactEmailAdress = loanApplication.Account.EmailAddress,
 
+            loanApplicationStatus = ApplicationStatus.Submitted.Value,
```
Comment (lukasz-ciura-pwc) [inline src/HE.InvestmentLoans.BusinessLogic/Application/Repositories/LoanApplicationRepository.cs:90]: This value should be setted by command handler, imho

File: src/HE.InvestmentLoans.BusinessLogic/User/UserAccount.cs [hunk 2]
```diff
@@ -1,3 +1,9 @@
 namespace HE.InvestmentLoans.BusinessLogic.User;
 
-public record UserAccount(string UserGlobalId, Guid AccountId);
+public record UserAccount(string UserGlobalId, Guid AccountId, string Name)
```
Comment (lukasz-ciura-pwc) [inline src/HE.InvestmentLoans.BusinessLogic/User/UserAccount.cs:3]: Name -> AccountName - just to be precisely

File: src/HE.InvestmentLoans.BusinessLogic/User/UserAccount.cs [hunk 3]
```diff
@@ -1,3 +1,9 @@
...
+{
+    public UserAccount(string userGlobalId, Guid accountId)
+        : this(userGlobalId, accountId, string.Empty)
```
Comment (lukasz-ciura-pwc) [inline src/HE.InvestmentLoans.BusinessLogic/User/UserAccount.cs:6]: I think we should not allow to pass empty AccountName

File: src/HE.InvestmentLoans.Contract/Application/Enums/ApplicationStatus.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
+namespace HE.InvestmentLoans.Contract.Application.Enums;
+
+public record ApplicationStatus(string Value)
```
Comment (lukasz-ciura-pwc) [inline src/HE.InvestmentLoans.Contract/Application/Enums/ApplicationStatus.cs:3]: Probably it should be in `ValueObjects` folder

File: src/HE.InvestmentLoans.Contract/Application/Enums/ApplicationStatus.cs [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+    public static ApplicationStatus NotApproved => new("not approved");
+
+    public static ApplicationStatus ApplicationDeclined => new("application declined");
+
+    public static ApplicationStatus ApprovedSubjectToContract => new("approved subject to contract");
+
+    public static bool operator ==(string value, ApplicationStatus status)
+    {
+        if (status is null)
```
Comment (lukasz-ciura-pwc) [inline src/HE.InvestmentLoans.Contract/Application/Enums/ApplicationStatus.cs:35]: Are you sure that an overriding equal operator is needed? Records have built-in functionality for it.

File: src/HE.InvestmentLoans.Contract/Application/Enums/ApplicationStatus.cs (hunk 4 above)
Comment (lukasz-ciura-pwc) [reply src/HE.InvestmentLoans.Contract/Application/Enums/ApplicationStatus.cs:3]: On other hand probably it should be an enum and mapping to crm values should be done in the data layer (repository)

**Review by dawid-pawulski — APPROVED** [review]

File: src/HE.InvestmentLoans.Contract/Application/ValueObjects/UserLoanApplication.cs [hunk 6]
```diff
@@ -1,27 +1,35 @@
...
+
+    public DateTime? LastModificationDate { get; }
 
     protected override IEnumerable<object> GetAtomicValues()
     {
         yield return Id;
         yield return ApplicationName;
         yield return Status;
+
+        if (LastModificationDate.HasValue)
```
Comment (lukasz-ciura-pwc) [inline src/HE.InvestmentLoans.Contract/Application/ValueObjects/UserLoanApplication.cs:30]: I think that if is not needed here

**Review by lukasz-ciura-pwc — APPROVED** [review]

**Pull request merged**